frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+const ROC_CURVE_DEFINITION =
+  'The receiver operating characteristic (ROC) curve shows the trade-off between true positive rate and false positive rate. ' +
+  'A lower threshold results in a higher true positive rate (and a higher false positive rate), ' +
+  'while a higher threshold results in a lower true positive rate (and a lower false positive rate)';
+
+type ConfidenceMetrics = {
+  confidenceThreshold: string;
+  falsePositiveRate: number;
+  recall: number;
+};
+
+type MetricsTabProps = {
+  execution: Execution;
+};
+
+interface MetricsSwitcherProps {
+  artifacts: Artifact[];
+  artifactTypes: ArtifactType[];
+}
+
+/**
+ * Metrics tab renders metrics for the artifact of given execution.
+ * Some system metrics are: Confusion Matrix, ROC Curve, Scalar, etc.
+ * Detail can be found in https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+ * Note that these metrics are only available on KFP v2 mode.
+ */
+export function MetricsTab({ execution }: MetricsTabProps) {
+  // Retriving a list of artifacts associated with this execution,
+  // so we can find the artifact for system metrics from there.
+  const { isLoading: isLoadingArtifacts, error: errorArtifacts, data: artifacts } = useQuery<
+    Artifact[],
+    Error
+  >(['execution_output_artifact', { id: execution.getId() }], () =>
+    getOutputArtifactsInExecution(execution),
+  );
+
+  // artifactTypes allows us to map from artifactIds to artifactTypeNames,
+  // so we can identify metrics artifact provided by system.
+  const {
+    isLoading: isLoadingArtifactTypes,
+    error: errorArtifactTypes,
+    data: artifactTypes,
+  } = useQuery<ArtifactType[], Error>(['artifact_types'], () => getArtifactTypes());
+
+  // This react element produces banner message if query to MLMD is pending or has error.
+  // Once query is completed, it shows actual content of metrics visualization in MetricsSwitcher.
+  const isLoading = isLoadingArtifactTypes && isLoadingArtifacts;
+  return (
+    <div className={commonCss.page}>
+      <div className={padding(20)}>
+        <div>
+          This step corresponds to execution{' '}
+          <Link
+            className={commonCss.link}
+            to={RoutePageFactory.executionDetails(execution.getId())}
+          >
+            "{ExecutionHelpers.getName(execution)}".
+          </Link>
+        </div>
+        {isLoading && <Banner message='Metrics is loading.' mode='info' />}
+        {errorArtifacts && (
+          <Banner
+            message='Error in retriving metrics information.'
+            mode='error'
+            additionalInfo={errorArtifacts.message}
+          />
+        )}
+        {errorArtifactTypes && (
+          <Banner
+            message='Error in retriving artifact types information.'
+            mode='error'
+            additionalInfo={errorArtifactTypes.message}
+          />
+        )}
+        {!isLoading && !errorArtifacts && !errorArtifactTypes && artifacts && artifactTypes && (
```

Comment:
note you also have status or isSuccess:
```
const {status, isSuccess} = useQuery(xxx)
status === 'success' # means the query result is already there
```
https://react-query.tanstack.com/reference/useQuery

Sounds like using `isSuccess` is the easiest here

Reply:
Totally! Updating.